pkg/telepresence/v1/Makefile

```diff
@@ -0,0 +1,31 @@
+# TELEPRESENCE_NAME is the name to use when creating intercepts.
+ifndef TELEPRESENCE_NAME
+$(error "TELEPRESENCE_NAME must be defined in the project's Makefile.")
+endif
+
+# TELEPRESENCE_PORT is the port to use when creating intercepts.
+ifndef TELEPRESENCE_PORT
+$(error "TELEPRESENCE_PORT must be defined in the project's Makefile.")
+endif
+
+# TELEPRESENCE_RUN_CMD is the command to use to start a local server for a
+# Telepresence intercept.
+TELEPRESENCE_RUN_CMD ?=
+
+# TELEPRESENCE_RUN_REQ is a space separated list of prerequisites needed to run
+# a local server for a Telepresence intercept.
+TELEPRESENCE_RUN_REQ +=
+
+################################################################################
+
+.PHONY: intercept
+intercept:
+	PORT=$(TELEPRESENCE_PORT) TELEPRESENCE_PORT=$(TELEPRESENCE_PORT) telepresence intercept $(TELEPRESENCE_NAME) --port $(TELEPRESENCE_PORT) -- $(shell echo $$SHELL)
+
+.PHONY: intercept-run
+intercept-run: $(TELEPRESENCE_RUN_REQ)
+ifndef TELEPRESENCE_RUN_CMD
+	$(error "TELEPRESENCE_RUN_CMD must be defined in the project's Makefile.")
+endif
+
+	PORT=$(TELEPRESENCE_PORT) TELEPRESENCE_PORT=$(TELEPRESENCE_PORT) telepresence intercept $(TELEPRESENCE_NAME) --port $(TELEPRESENCE_PORT) -- $(TELEPRESENCE_RUN_CMD)
```

Comment:
Personally I would still be inclined to leave these environment variables out, thinking as follows:

- There's no scenario where we can actually make use of this without making our apps aware of `TELEPRESENCE_PORT`, and I'd seek to avoid that
- Anywhere we use `PORT` in practice it would be overridden by `telepresence intercept` anyway
- It makes it appear as though they're required/important when infact they do nothing (red herring when debugging?)

I _could_ see some value in setting `PORT` in the shell if we could work out how to do that, though only for some hypothetical open source project.

I don't feel particularly strongly about this.

Reply:
The `PORT` variable _does_ make it into the sub-shell using this method. I'm also not sure what you mean by it being overridden by `telepresence intercept` - Telepresence doesn't set any kind of port variables currently.

Reply:
(aside from the specific ones for services, but those don't clash here)

Reply:
It makes it in there because it's not in our manifests. If it were in our manifests (as it should be, if we were using it) it would be overridden by `telepresence intercept` with the value from the manifest.

Reply:
Yeah, okay, I get what you mean now. In a world where we don't have any optional env vars, it makes little sense.